test/declarative_test_fwk/runner.py

```diff
@@ -0,0 +1,226 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+#	 http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+"""Runner for Declarative tests framework scenarios for custom resources
+"""
+
+from declarative_test_fwk import model, helper
+import pytest
+import sys
+import logging
+from time import sleep
+from acktest.k8s import resource as k8s
+
+
+def run(scenario: model.Scenario) -> None:
+    """Runs steps in the given scenario
+
+    Args:
+        scenario: the scenario to run
+
+    Returns:
+        None
+    """
+
+    logging.info(f"Execute: {scenario}")
+    for step in scenario.steps():
+        run_step(step)
+
+
+def teardown(scenario: model.Scenario) -> None:
+    """Teardown steps in the given scenario in reverse run order
+
+    Args:
+        scenario: the scenario to teardown
+
+    Returns:
+        None
+    """
+
+    logging.info(f"Teardown: {scenario}")
+    teardown_failures = []
+    # tear down steps in reverse order
+    for step in reversed(scenario.steps()):
+        try:
+            teardown_step(step)
+        except:
+            error = f"Failed to teardown: {step}. " \
+                    f"Unexpected error: {sys.exc_info()[0]}"
+            teardown_failures.append(error)
+            logging.debug(error)
```

Comment:
do we need this `logging.debug` line if we're using `logging.error` below on line 65?

Reply:
Removed debug statement.